actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -157,6 +139,81 @@ def distance_of_time_in_words(from_time, to_time = 0, options = {})
         end
       end
 
+      def distance_of_dates_in_words_simple_example(from:, to:)
+        distance = DateDistance.new(from, to)
+
+        duration = if distance.in_days < 1.0
+          { key: :x_hours, count: [distance.in_hours.round, 1].min }
+        elsif distance.in_days < 30.0
+          { key: :x_days, count: distance.in_days.round }
+        elsif distance.in_years < 1.0
+          { key: :x_months, count: distance.in_months.round }
+        else
+          { key: :x_years, count: distance.in_years.round }
+        end
+
+        I18n.with_options scope: :'datetime.distance_in_words' do |locale|
+          locale.t(duration[:key], count: duration[:count])
+        end
+      end
+
+      class DateDistance
+        attr_reader :from_time, :to_time, :in_minutes, :in_seconds
+
+        def initialize(from_time, to_time)
+          @from_time = normalize_distance_of_time_argument_to_time(from_time)
+          @to_time = normalize_distance_of_time_argument_to_time(to_time)
+          @from_time, @to_time = @to_time, @from_time if @from_time > @to_time
+          @in_minutes = ((@to_time - @from_time) / 60.0).round
+          @in_seconds = (@to_time - @from_time).round
```

Comment:
We could probably remove the rounding here to fix some of the bugs mentioned in https://github.com/rails/rails/issues/21916, but would have to add rounding where we've checked `distance.in_minutes == 0` and where we've used it directly as a count.

This would then be more consistent with the methods for hours/days/months/years where we expose a float